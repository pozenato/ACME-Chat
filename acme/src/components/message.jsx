import React, { Component } from 'react';
import moment from 'moment';


export default class Message extends Component {

    constructor(props) {
        super()
    }

    handleSameMessage = (rightDirection, message) => (
        <div className="messageDisplay">
                <div className={rightDirection ? "right messageText" : "left messageText"}>{message.text}</div>
        </div> 
    )

    handleNewAuthorMessage = (rightDirection, message) => (
        <div className="messageDisplay">
                <div className={rightDirection ? "rightHeader messageHeader" : "leftHeader messageHeader"}>{<img className="messageImage" src={message.userMessage.image} alt="logo img"/>} {message.userMessage.name}, {moment(message.date).format('LT')}</div>
                <div className={rightDirection ? "right messageText" : "left messageText"}>{message.text}</div>
        </div>
    )

    render() {
        const {
            message,
            sameAuthor,
            sameHour,
            rightDirection,
        } = this.props;
        return (
            <div className="messageContext" id="inner">
                {
                    sameHour && sameAuthor ?
                        this.handleSameMessage(rightDirection, message)
                    :
                        this.handleNewAuthorMessage(rightDirection, message)
                }
            </div>

        )
    }
};