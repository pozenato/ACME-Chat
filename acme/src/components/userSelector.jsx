import React, { Component} from 'react';
import { Menu, Dropdown, Icon, Button, message } from 'antd';


class UserSelector extends Component {

  constructor(props) {
    super()
  }

  handleButtonClick = (e) => {
    message.info('Click on left button.');
  }

  handleMenuClick = (e) => {
    this.props.handleUserSelected(e.key)
  }

  menuItems = () => {
    if (this.props.users) {
      return (
        <Menu onClick={(this.handleMenuClick)} className="menuConteiner">
          {
            this.props.users.map(u => {
              return (
                <Menu.Item className="userItemMenu" key={u.id}>
                      <img className="userImage" src={u.image} alt="logo img"/>
                      <h4 className="nameItem">{u.name}</h4>
                </Menu.Item>
              )
            })
          }
        </Menu>
      )
    }
    return <Menu />;
  };

render() {

    const { userSelected, defaultText } = this.props;

    return (
      <div id="components-dropdown-userSelector-dropdown-button">
        <Dropdown className="DropDownMenu" overlay={this.menuItems()}>
          <Button>
            {
              userSelected.id > 0 ? 
              <div className="userItemMenu">
                <img className="userImage" src={userSelected.image} alt="logo img"/>
                  <h4 className="nameItem">{userSelected.name}</h4>
              </div> : 
              <div>
                {defaultText}
                <Icon className="searchIcon" type="down" />
              </div>
            }            
          </Button>
        </Dropdown>
      </div>
    );
  }
}

export default UserSelector;