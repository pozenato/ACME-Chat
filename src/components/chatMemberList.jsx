import React, { Component } from 'react'

class chatMemberList extends Component {

    constructor(props) {
        super()
    }

    fetchChatMembers = () => {
        let unique = null;
        let messageUsers = [];
        if (this.props.messages) {
            messageUsers = this.props.messages.map(m => m.userMessage);
            unique = [ ...new Set( messageUsers ) ];
        }
        return unique;
    }

    render() {
        const users = this.fetchChatMembers();

        return (
            <div>
                <h1 className="chatMemberTitle">Chat Members</h1>
                {
                    users !== null ?
                        users.map(m => (
                            <div className="chatMemberContext" key={m.id}>
                                <img className="chatMemberImage" src={m.image} alt="logo img"/>
                                <h4 className="chatMemberName">{m.name}</h4>
                            </div>
                        )) 
                    : 
                        null
                }
            </div>
        )
    }
}

export default chatMemberList;