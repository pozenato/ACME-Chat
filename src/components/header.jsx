import React from 'react';
import logo from '../../resources/images/Acme-corp.png';

const Header = () => (
    <div className="headerContainer">
        <div className="logoContent">
            <img className="headerImg" src={logo} alt="logo img"/>            
        </div>
        <div className="titleContent">
            <h1 className="headerText">ACME Discussions</h1>
        </div>
    </div>
)

export default Header;