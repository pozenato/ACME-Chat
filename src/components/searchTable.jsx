import React from 'react'
import { Table } from 'antd';

export default props => {

    const columns = [
        {
          title: 'Name',
          dataIndex: 'name',
          key: 'name',
        },
        {
          title: 'Date',
          dataIndex: 'date',
          key: 'date',
        },
        {
          title: 'Message',
          dataIndex: 'message',
          key: 'message',
        }
    ];
    
    return (
        <Table columns={columns} dataSource={props.messages} />
    );
}