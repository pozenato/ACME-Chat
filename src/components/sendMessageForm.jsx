import React, { Component } from 'react';
import component from '../components';
import { Input, Button } from 'antd';
import moment from 'moment';
import findUserSelected from '../utils/sharedFunctions';
import constants from '../utils/constants';


class SendMessageForm extends Component {

    constructor(props) {
        super(props)
        this.state = this.getInitialState()
    }

    getInitialState() {
        return {
            userSelected: 0,
            messageValue: '',
        }
    }

    handleUserSelected = user => {
        const userSelected = findUserSelected(user, this.props.users);
        this.setState({ userSelected });
    }

    handleSubmitMessage = () => {
        this.props.addNewMessage({            
            user: this.state.userSelected, 
            date: moment(),
            text: this.state.messageValue,
        }
        );
        this.setState({ messageValue: '' })
    };

    handleSubmit = (event) => {
        event.preventDefault();
        this.handleSubmitMessage();
    }

    render() {
        const {
            userSelected,
            messageValue,
        } = this.state;
        const {
            users,
        } = this.props;
        return (
            <form onSubmit={this.handleSubmit}>
                <div className="sendMessageContext">
                    <div className="selectorMessage">
                        <component.userSelector
                            userSelected={userSelected}
                            handleUserSelected={this.handleUserSelected}
                            users={users}
                            defaultText={constants.userSelector.sendMessageDefaultText}
                        />         
                    </div>
                    <Button shape="circle" type="primary"  icon="search" onClick={this.props.showModal} />
                    <div className="inputMessage">
                        <Input
                            placeholder="Type a Message"
                            disabled={userSelected === 0}
                            value={messageValue}
                            onChange={e => this.setState({ messageValue: e.target.value })} />
                    </div>
                    <div className="sendButton">
                        <Button
                            type="primary"
                            shape="round"
                            icon="rocket"
                            size='large'
                            disabled={messageValue === ""}
                            onClick={this.handleSubmitMessage}
                        >
                            Send
                        </Button>
                    </div>
                </div>
            </form>
        );
    };

}

export default SendMessageForm;