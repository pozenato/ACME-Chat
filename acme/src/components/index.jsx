import chatMemberList from './chatMemberList';
import message from './message';
import messageList from './messageList';
import sendMessageForm from './sendMessageForm';
import userSelector from './userSelector';
import header from './header';
import searchTable from './searchTable';

export default {
    chatMemberList,
    message,
    messageList,
    sendMessageForm,
    userSelector,
    header,
    searchTable,
}