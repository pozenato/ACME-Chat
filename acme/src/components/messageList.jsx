import React, { Component } from 'react'
import components from '../components'

export default class MessageList extends Component {

    constructor(props) {
        super()
    }

    compareMinuteswithMessages = (currentDate, lastMessage) => {
        const minutes = (currentDate.diff(lastMessage.date, 'minutes'));
        return minutes < 1 ? true : false;
    }

    render() {
        const { messages } = this.props;
        let sameAuthor, sameHour = false;
        let rightDirection = true;
        let lastMessage = null;
        return (
            <div>
                {messages.map(m => {
                    if (lastMessage !== null) {
                        sameAuthor = m.userMessage.name === lastMessage.userMessage.name || false;
                        if (sameAuthor) {
                            sameHour = this.compareMinuteswithMessages(m.date, lastMessage);
                        } else {
                            sameHour = false;
                            rightDirection = !rightDirection;
                        }
                    }
                    lastMessage = m;
                    return (
                        <components.message
                            key={m.id}
                            message={m}
                            sameAuthor={sameAuthor}
                            sameHour={sameHour}
                            rightDirection={rightDirection}
                        />
                        )
                    }
                )}
            </div>
        )
    }
}