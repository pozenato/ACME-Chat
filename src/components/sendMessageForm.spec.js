import React from 'react';
import components from '../components'
import renderer from 'react-test-renderer'

describe("The Send Message Form", ()=>{
    it("renders as expected", ()=> {
        const tree = renderer
        .create(<components.sendMessageForm tags={[`css`, `html`, `go`]}/>)
        .toJSON();
        
        expect(tree).toMatchSnapshot();
    })
})