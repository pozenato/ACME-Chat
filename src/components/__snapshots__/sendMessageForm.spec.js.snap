// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Send Message Form renders as expected 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="sendMessageContext"
  >
    <div
      className="selectorMessage"
    >
      <div
        id="components-dropdown-userSelector-dropdown-button"
      >
        <button
          className="ant-btn ant-dropdown-trigger DropDownMenu"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          type="button"
        >
          <div>
            Choise an User to send a Message
            <i
              aria-label="icon: down"
              className="anticon anticon-down searchIcon"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </i>
          </div>
        </button>
      </div>
    </div>
    <button
      className="ant-btn ant-btn-primary ant-btn-circle ant-btn-icon-only"
      onClick={[Function]}
      type="button"
    >
      <i
        aria-label="icon: search"
        className="anticon anticon-search"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="search"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
          />
        </svg>
      </i>
    </button>
    <div
      className="inputMessage"
    >
      <input
        className="ant-input ant-input-disabled"
        disabled={true}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Type a Message"
        type="text"
        value=""
      />
    </div>
    <div
      className="sendButton"
    >
      <button
        className="ant-btn ant-btn-primary ant-btn-round ant-btn-lg"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        <i
          aria-label="icon: rocket"
          className="anticon anticon-rocket"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="rocket"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M864 736c0-111.6-65.4-208-160-252.9V317.3c0-15.1-5.3-29.7-15.1-41.2L536.5 95.4C530.1 87.8 521 84 512 84s-18.1 3.8-24.5 11.4L335.1 276.1a63.97 63.97 0 0 0-15.1 41.2v165.8C225.4 528 160 624.4 160 736h156.5c-2.3 7.2-3.5 15-3.5 23.8 0 22.1 7.6 43.7 21.4 60.8a97.2 97.2 0 0 0 43.1 30.6c23.1 54 75.6 88.8 134.5 88.8 29.1 0 57.3-8.6 81.4-24.8 23.6-15.8 41.9-37.9 53-64a97 97 0 0 0 43.1-30.5 97.52 97.52 0 0 0 21.4-60.8c0-8.4-1.1-16.4-3.1-23.8H864zM762.3 621.4c9.4 14.6 17 30.3 22.5 46.6H700V558.7a211.6 211.6 0 0 1 62.3 62.7zM388 483.1V318.8l124-147 124 147V668H388V483.1zM239.2 668c5.5-16.3 13.1-32 22.5-46.6 16.3-25.2 37.5-46.5 62.3-62.7V668h-84.8zm388.9 116.2c-5.2 3-11.2 4.2-17.1 3.4l-19.5-2.4-2.8 19.4c-5.4 37.9-38.4 66.5-76.7 66.5-38.3 0-71.3-28.6-76.7-66.5l-2.8-19.5-19.5 2.5a27.7 27.7 0 0 1-17.1-3.5c-8.7-5-14.1-14.3-14.1-24.4 0-10.6 5.9-19.4 14.6-23.8h231.3c8.8 4.5 14.6 13.3 14.6 23.8-.1 10.2-5.5 19.6-14.2 24.5zM464 400a48 48 0 1 0 96 0 48 48 0 1 0-96 0z"
            />
          </svg>
        </i>
        <span>
          Send
        </span>
      </button>
    </div>
  </div>
</form>
`;
